statsmodels/stats/anova.py

```diff
@@ -362,6 +367,197 @@ def anova_lm(*args, **kwargs):
+        error_message = "Data is unbalanced."
+        if len(cell_count) != factor_levels:
+            raise ValueError(error_message)
+        count = cell_count[key]
+        for key in cell_count:
+            if count != cell_count[key]:
+                raise ValueError(error_message)
+        if self.data.shape[0] > count * factor_levels:
+            raise ValueError('There are more than 1 element in a cell! Missing'
+                             ' factors?')
+
+    def fit(self):
+        y = self.data[self.dv].values
+
+        # Construct OLS endog and exog from string using patsy
+        within = ['C(%s, Sum)' % i for i in self.within]
+        subject = 'C(%s, Sum)' % self.subject
+        factors = within + [subject]
+        x = patsy.dmatrix('*'.join(factors), data=self.data)
+        term_slices = x.design_info.term_name_slices
+        for key in term_slices:
+            ind = np.array([False]*x.shape[1])
+            ind[term_slices[key]] = True
+            term_slices[key] = np.array(ind)
+        term_exclude = [':'.join(factors)]
+        ind = _not_slice(term_slices, term_exclude, x.shape[1])
+        x = x[:, ind]
+
+        # Fit OLS
+        model = OLS(y, x)
+        results = model.fit()
+        if model.rank < x.shape[1]:
+            raise ValueError('Independent variables are collinear.')
+        for i in term_exclude:
+            term_slices.pop(i)
+        for key in term_slices:
+            term_slices[key] = term_slices[key][ind]
+        params = results.params
+        df_resid = results.df_resid
+        ssr = results.ssr
+
+        anova_table = pd.DataFrame(
+            {'F Value': [], 'Num DF': [], 'Den DF': [], 'Pr > F': []})
+
+        for key in term_slices:
+            if self.subject not in key and key != 'Intercept':
+                #  Independen variables are orthogonal
+                ssr1, df_resid1 = _ssr_reduced_model(
+                    y, x, term_slices, params, [key])
+                df1 = df_resid1 - df_resid
+                msm = (ssr1 - ssr) / df1
+                if (key == ':'.join(factors[:-1]) or
+                        (key + ':' + subject not in term_slices)):
+                    mse = ssr / df_resid
+                    df2 = df_resid
+                else:
+                    ssr1, df_resid1 = _ssr_reduced_model(
+                        y, x, term_slices, params,
+                        [key + ':' + subject])
+                    df2 = df_resid1 - df_resid
+                    mse = (ssr1 - ssr) / df2
+                F = msm / mse
+                p = stats.f.sf(F, df1, df2)
+                term = key.replace('C(', '').replace(', Sum)', '')
+                anova_table.loc[term, 'F Value'] = F
+                anova_table.loc[term, 'Num DF'] = df1
+                anova_table.loc[term, 'Den DF'] = df2
+                anova_table.loc[term, 'Pr > F'] = p
+
+        return ANOVAResults(anova_table.iloc[:, [1, 2, 0, 3]])
+
+
+class ANOVAResults(object):
```

Comment:
I think this should better be renamed to `AnovaResults` to be consistent with `AnovaRM`. Also looks prettier, imho. ;)